ecommerce-ui/src/Listings.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import bnbs from './bnbs.json';
+import Oops from './Oops';
+// import Stars from './Stars';
+// import SuperHost from './SuperHost';
+
+export default function Listings({ onAdd, cartItems }) {
+  return (
+    <>
+      {bnbs.map((listing) => (
+          <div key={listing.id} className="card Card-width m-2 p-2">
+            <img src={listing.image} className="card-img-top" alt={listing.title} />
+            <div className="card-body">
+              <a href={listing.credit} className="Host-info">Image credit</a>
+              <h4 className="card-title">{listing.title}</h4>
+              <div className="card-text fw-bold">
+                {listing.houseType} in {listing.location.city} {listing.location.country}</div>
+              <div className="card-text">Your host: {listing.host.name}</div>
+              <div className="card-text">${listing.payment.cost}</div>
+              <div>{listing.payment.description}</div>
+              {/* <Stars rating={listing.rating.stars} /> */}
+            </div>
+            <div className="card-footer">
+              <button className="btn btn-primary" onClick={() => cartItems.includes(listing) ? Oops() : onAdd(listing)} >Add to cart</button>
+              {/* <SuperHost prop={listing.host.isSuperhost} /> */}
+            </div>
+          </div>
+      ))}
+    </>
+  );
+}
+
+Listings.propTypes = {
+  id: PropTypes.string.isRequired,
+  image: PropTypes.string.isRequired,
+  title: PropTypes.string.isRequired,
+  houseType: PropTypes.string.isRequired,
+  location: PropTypes.shape({
+    city: PropTypes.string.isRequired,
+    country: PropTypes.string.isRequired,
+  }),
+  payment: PropTypes.shape({
+    cost: PropTypes.string.isRequired,
+    description: PropTypes.string.isRequired,
+  }),
```

Comment:
Note that you'll just need the propTypes for `cartItems` and `onAdd`, since the rest of the attributes are coming from the bnbs object (i.e. not props)

Reply:
I see why now. Thank you!